### ISSUE TITLE
Trying to search for VMs in a specified folder is not working

### ISSUE THREAD

**Issue by lbrandel** [chef-partners/knife-vsphere#314]
### What are you trying to do and what's not working?

I'm trying to use the "knife vsphere vm find" command with the -f or --folder option to list the VMs in a particular folder.  The output does not seem to filter on a given folder name, it just returns all the VMs in the given cluster.  Is that even a valid use case?  From the usage menu when I just provide "knife vsphere vm find" it would suggest that it should work, but it is not.

Looking at the documentation, it appears perhaps only --match-name is supported.  Is that correct?  If so, perhaps the usage output should not show the -f/--folder option when trying to do a "knife vsphere vm find"

### Other details, such as gem version and if you're working with a Linux or Windows VM

I'm running the knife vsphere command on both a Mac and Linux client.  Same issue.

**Comment by swalberg** [comment]
Hi, the code for the find command doesn't use the folder option. You see the folder option because the base class adds it as an option, so every command gets it.

Your comment about not showing it here makes sense, and I'll tag this as something to fix.

Have a look through the [code](https://github.com/chef-partners/knife-vsphere/blob/master/lib/chef/knife/vsphere_vm_find.rb) to see all the options. Does `--match-name` work for you? 

If you want to see all the vms in a folder, there is also `knife vsphere vm list` which does support folders.

**Comment by lbrandel** [comment]
Yes, `--match-name` does work for me, so that is a help.  I'll also investigate `knife vsphere vm list`
Thank you!
